scanner/Makefile

```diff
@@ -19,15 +19,39 @@ else
 QUAY_TAG_EXPIRATION=never
 endif
 
+# CGO is disabled for Scanner V4 builds because:
+#
+# 1. Local builds aren't dockerized. When e.g. host OS's glibc does not match version in target container, scanner
+# refuses to start with errors looking like this:
+#    $ /usr/local/bin/scanner --help
+#    /usr/local/bin/scanner: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /usr/local/bin/scanner)
+#    /usr/local/bin/scanner: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /usr/local/bin/scanner)
+#
+# 2. GitHub builds for arm64, ppc64le and s390x need support for cross-compilation which, when unavailable, makes
+# builds fail with errors like
+#    gcc_arm64.S:30: Error: no such instruction: `stp x29,x30,[sp,'
+#    gcc_arm64.S:34: Error: too many memory references for `mov'
+#
+# This flag must be overridden in commercial/downstream builds in arguments to `make`.
+#    make CGO_ENABLED=1 blah          # Correct - flips it to 1.
+#    CGO_ENABLED=1 make blah          # Wrong! This has no effect. Don't do this.
+#    export CGO_ENABLED=1; make blah  # Wrong! This has no effect. Don't do this.
+# Similarly, don't do the following in Dockerfile
+#    ENV CGO_ENABLED=1                # Wrong!..
+#    RUN make blah                    # because it has no effect. Don't do this.
+# If you need to separate the declaration and the usage in Dockerfile, do this instead
+#    ARG CGO_ENABLED=1
+#    RUN make CGO_ENABLED=$CGO_ENABLED blah
```

Comment:
Feels like this part of the comment does not belong here. It seems related to patterns and good practices for building Docker images. Maybe it belongs to the related `Dockerfiles` or separate documentation such as `README.md`?

Reply:
I'd go even further to say it's a basic knowledge that does not deserve any explicit mention. Still, I met
```Dockerfile
ENV CGO_ENABLED=1
RUN make ... CGO_ENABLED=${CGO_ENABLED} image/scanner/bin/scanner ...
```
and was wondering what process subtleties am **I** not getting that require setting the environment variable in addition to overriding the Makefile variable.

I took it as this was intentional and undocumented which took me to a rabbit hole with this PR whereas it was likely unintentional. If my comment helps to prevent at least in one such instance, the effort I made to write, you made to review and other engineers to read it will be justified, in my opinion.
Therefore, I'd keep it unless you still feel it's a bad idea.

Reply:
The comment's content is valuable, as is the process of writing it, sharing the information, and its usefulness to prevent confusion. What I am questioning is the place where it is being published, and if it would not be more beneficial to its purpose to place it closer to the context where it is applicable (the downstream Dockerfile) rather than here.

But I don't think it is a "bad idea", I just think it belongs somewhere else. At this point I think this is "nitpicking", so feel free to resolve this comment and move on. Thank you!

Oh, by the way, I think GitHub cannot show exactly the part that I highlighted in my [original comment](https://github.com/stackrox/stackrox/pull/10853/files#r1583886450)--I am talking only about this part:

```make
# This flag must be overridden in commercial/downstream builds in arguments to `make`.
#    make CGO_ENABLED=1 blah          # Correct - flips it to 1.
#    CGO_ENABLED=1 make blah          # Wrong! This has no effect. Don't do this.
#    export CGO_ENABLED=1; make blah  # Wrong! This has no effect. Don't do this.
# Similarly, don't do the following in Dockerfile
#    ENV CGO_ENABLED=1                # Wrong!..
#    RUN make blah                    # because it has no effect. Don't do this.
# If you need to separate the declaration and the usage in Dockerfile, do this instead
#    ARG CGO_ENABLED=1
```

Reply:
I think I understood you. I should have mentioned the reason I did not consider moving it to the downstream Dockerfile: this Makefile is called not just from one Dockerfile but from many places: (1) downstream Dockerfile, (2) Konflux Dockerfile, (3) GitHub Actions. If the comment pops up in one place, two others remain without it and that would be a knowledge easier to miss. If I duplicate the comment in three places, well, that's even worse. Therefore I put it here using the analogy that CGO_ENABLED is an API and this comment is API doc.